src/pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from 'react'
+import styles from './MainPage.module.scss'
+import { Header } from '@components/Header/Header'
+import { Image } from '@components/Image/Image'
+import { useImages } from '@/features/images/useImages'
+import { Button } from '@components/Button/Button'
+import { LinkComponent } from '@/components/LinkComponent/LinkComponent'
+import { PAINT_PAGE } from '@/variables/routes'
+import { typeImages, typeImage } from '@/repositories/images/interfaces/imagesController'
+
+enum TypesCards {
+  All,
+  Filtered,
+}
+
+export const MainPage = () => {
+  const { images } = useImages()
+
+  const [cards, setCards] = useState<typeImages | null>(null)
+  const [typeCards, setTypeCard] = useState<TypesCards>(TypesCards.All)
+
+  useEffect(() => {
+    setCards(images)
+  }, [images])
+
+  const sortCards = (email: string | undefined) => {
+    switch (typeCards) {
+      case TypesCards.All: {
+        if (!email) return
+
+        const filteredCards = images?.filter((card: typeImage) => card.user === email)
```

Comment:
У тебя images может быть undefined. В чем смысл выполнять сортировку если у тебя не будет images ?

```suggestion
        if (!email || !images) return

        const filteredCards = images.filter((card: typeImage) => card.user === email)
```